packages/dds/merge-tree/src/test/testUtils.ts

```diff
@@ -26,30 +27,100 @@ export function loadTextFromFileWithMarkers(filename: string, mergeTree: MergeTr
     return loadText(content, mergeTree, segLimit, true);
 }
 
-export function insertMarker(
-    mergeTree: MergeTree,
-    pos: number,
-    refSeq: number,
-    clientId: number,
-    seq: number,
-    behaviors: ReferenceType, props: PropertySet | undefined, opArgs: IMergeTreeDeltaOpArgs,
-) {
+interface InsertMarkerArgs {
+    mergeTree: MergeTree;
+    pos: number;
+    refSeq: number;
+    clientId: number;
+    seq: number;
+    behaviors: ReferenceType;
+    props: PropertySet | undefined;
+    opArgs: IMergeTreeDeltaOpArgs;
+}
+
+export function insertMarker({
+    mergeTree,
+    pos,
+    refSeq,
+    clientId,
+    seq,
+    behaviors,
+    props,
+    opArgs,
+}: InsertMarkerArgs) {
     mergeTree.insertSegments(pos, [Marker.make(behaviors, props)], refSeq, clientId, seq, opArgs);
 }
 
-export function insertText(
-    mergeTree: MergeTree,
-    pos: number,
-    refSeq: number,
-    clientId: number,
-    seq: number,
-    text: string,
-    props?: PropertySet,
-    opArgs?: IMergeTreeDeltaOpArgs,
-) {
+interface InsertTextArgs {
+    mergeTree: MergeTree;
+    pos: number;
+    refSeq: number;
+    clientId: number;
+    seq: number;
+    text: string;
+    props?: PropertySet;
+    opArgs?: IMergeTreeDeltaOpArgs;
+}
+
+export function insertText({
+    mergeTree,
+    pos,
+    refSeq,
+    clientId,
+    seq,
+    text,
+    props,
+    opArgs,
+}: InsertTextArgs) {
     mergeTree.insertSegments(pos, [TextSegment.make(text, props)], refSeq, clientId, seq, opArgs);
 }
 
+interface InsertSegmentsArgs {
+    mergeTree: MergeTree;
+    pos: number;
+    segments: ISegment[];
+    refSeq: number;
+    clientId: number;
+    seq: number;
+    opArgs: IMergeTreeDeltaOpArgs | undefined;
+}
+
+export function insertSegments({
+    mergeTree,
+    pos,
+    segments,
+    refSeq,
+    clientId,
+    seq,
+    opArgs,
+}: InsertSegmentsArgs): void {
```

Comment:
this is an interesting pattern. i've never seen an object expanded in a params list like this before. neat.